homeassistant/helpers/entity_platform.py

```diff
@@ -269,10 +272,22 @@ def add_entities(self, new_entities, update_before_add=False):
             else:
                 config_entry_id = None
 
+            if entity.device is not None:
```

Comment:
`entity.device` will actually execute code so better to just request it once instead of 8 times.

```python
device = entity.device

if device is not None:
```